main/src/main/java/com/sedmelluq/discord/lavaplayer/source/nico/NicoAudioTrack.java

```diff
@@ -2,35 +2,46 @@
 
 import com.sedmelluq.discord.lavaplayer.container.mpeg.MpegAudioTrack;
 import com.sedmelluq.discord.lavaplayer.source.AudioSourceManager;
+import com.sedmelluq.discord.lavaplayer.tools.JsonBrowser;
 import com.sedmelluq.discord.lavaplayer.tools.io.HttpClientTools;
 import com.sedmelluq.discord.lavaplayer.tools.io.HttpInterface;
 import com.sedmelluq.discord.lavaplayer.tools.io.PersistentHttpStream;
 import com.sedmelluq.discord.lavaplayer.track.AudioTrack;
 import com.sedmelluq.discord.lavaplayer.track.AudioTrackInfo;
 import com.sedmelluq.discord.lavaplayer.track.DelegatedAudioTrack;
 import com.sedmelluq.discord.lavaplayer.track.playback.LocalAudioTrackExecutor;
+import org.apache.http.HttpStatus;
 import org.apache.http.client.methods.CloseableHttpResponse;
 import org.apache.http.client.methods.HttpGet;
-import org.apache.http.client.utils.URLEncodedUtils;
-import org.apache.http.util.EntityUtils;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.net.URI;
 import java.nio.charset.StandardCharsets;
-import java.util.Map;
-
-import static com.sedmelluq.discord.lavaplayer.tools.DataFormatTools.convertToMapLayout;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
 
 /**
  * Audio track that handles processing NicoNico tracks.
  */
 public class NicoAudioTrack extends DelegatedAudioTrack {
     private static final Logger log = LoggerFactory.getLogger(NicoAudioTrack.class);
 
+    private static final ScheduledExecutorService executorService = Executors.newSingleThreadScheduledExecutor();
```

Comment:
creating one executor service per track seems rather excessive, can't this be handled in the source manager instead?

Reply:
I'm not sure why this heartbeat thing is needed, or what it even does
the blog post is in japanese after all

Reply:
Discussed within the Lavalink server, I clarified this was a static instance so only one thread for all tracks. However the question of "is this necessary?" does remain

Reply:
It appears to be necessary from when I've tested it - without one request at least every 2 minutes, the video stream 403's.